alacritty/src/display.rs

```diff
@@ -358,16 +368,87 @@ impl Display {
     /// This call may block if vsync is enabled
     pub fn draw<T>(
         &mut self,
-        terminal: MutexGuard<'_, Term<T>>,
+        mut terminal: MutexGuard<'_, Term<T>>,
         message_buffer: &MessageBuffer,
         config: &Config,
+        event_queue: &mut Vec<GlutinEvent<Event>>,
     ) {
         let grid_cells: Vec<RenderableCell> = terminal.renderable_cells(config).collect();
+        let visual_bell_animating = terminal.visual_bell.animating();
         let visual_bell_intensity = terminal.visual_bell.intensity();
         let background_color = terminal.background_color();
         let metrics = self.glyph_cache.font_metrics();
         let glyph_cache = &mut self.glyph_cache;
         let size_info = self.size_info;
+        let cols = size_info.cols();
+
+        // Request immediate re-draw if visual bell animation is not finished
+        if visual_bell_animating {
+            event_queue.push(GlutinEvent::UserEvent(Event::Wakeup));
```

Comment:
Could we just pass `bell_animating: bool` true instead of &mut ref to `event_queue`? I don't like the idea, that event_queue is in `draw` method for literally no reason.

Reply:
I'll left this one to @chrisduerr